library/src/main/java/com/twotoasters/sectioncursoradapter/adapter/datahandler/ArrayDataHandler.java

```diff
@@ -0,0 +1,370 @@
+package com.twotoasters.sectioncursoradapter.adapter.datahandler;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.RecyclerView.Adapter;
+import android.widget.Filter;
+import android.widget.Filterable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * For full animation support {@link #setAdapter(Adapter)}.
+ *
+ * {@inheritDoc}
+ */
+@SuppressWarnings("unused")
+public class ArrayDataHandler<T> implements DataHandler<T>, Filterable {
+    @Nullable private Adapter<?> mAdapter;
+
+    /**
+     * Contains the list of objects that represent the data of this ArrayAdapter.
+     * The content of this list is referred to as "the array" in the documentation.
+     */
+    @NonNull private List<T> mObjects = new ArrayList<>();
+    /**
+     * Lock used to modify the content of {@link #mObjects}. Any write operation
+     * performed on the array should be synchronized on this lock. This lock is also
+     * used by the filter (see {@link #getFilter()} to make a synchronized copy of
+     * the original array of data.
+     */
+    private final Object mLock = new Object();
+    // A copy of the original mObjects array, initialized from and then used instead as soon as
+    // the mFilter ArrayFilter is used. mObjects will then only contain the filtered values.
+    private ArrayList<T> mOriginalValues;
+    private ArrayFilter mFilter;
+
+    /**
+     * Constructor
+     */
+    public ArrayDataHandler() {
+        init(null, null);
+    }
+
+    /**
+     * Constructor
+     *
+     * @param objects The objects to represent in the ListView. These will be copied so that outside modification is ignored.
```

Comment:
I think so. I don't really know what I could do about that. I wanted to make sure that changes in the array size are hitting the adapter correctly.